src/scripts/main.js

```diff
@@ -1,3 +1,37 @@
 'use strict';
 
 // write code here
+
+const employeeList = document.querySelector('ul');
+
+function sortList(list) {
+  const items = Array.from(list.children);
+
+  items.sort((a, b) => {
+    const salaryA = parseSalary(a.getAttribute('data-salary'));
+    const salaryB = parseSalary(b.getAttribute('data-salary'));
+
+    return salaryB - salaryA;
+  });
+
+  items.forEach((item) => list.appendChild(item));
+}
+
+function parseSalary(salaryStr) {
+  return parseInt(salaryStr.replace(/[^0-9.-]+/g, ''));
+}
+
+function getEmployees(list) {
+  return Array.from(list.children).map((item) => {
+    return {
+      name: item.textContent.trim(),
+      position: item.getAttribute('data-position'),
+      salary: parseSalary(item.getAttribute('data-salary')),
+      age: parseInt(item.getAttribute('data-age')),
```

Comment:
Ensure that the `data-age` attribute is always present and correctly formatted as an integer. If there's a possibility of missing or malformed data, consider adding error handling or default values.